mesop/components/uploader/uploader.py

```diff
@@ -0,0 +1,72 @@
+import json
+from dataclasses import dataclass
+from typing import Any, Callable
+
+import mesop.components.uploader.uploader_pb2 as uploader_pb
+from mesop.component_helpers import (
+  insert_component,
+  register_event_handler,
+  register_event_mapper,
+  register_native_component,
+)
+from mesop.events import MesopEvent
+
+
+@dataclass(kw_only=True)
+class UploadChangeEvent(MesopEvent):
+  """Event representing a change in the uploader component's value.
+
+  Attributes:
+      name: The file's name.
+      size: The size of the file in bytes.
+      type: The file's MIME type.
+      data_url: Uploaded content as data URL.
```

Comment:
One of the main value propositions with Mesop is that it interoperates well with the broader Python ecosystem (e.g. the table component directly accepts Pandas dataframe). In this case, I think returning a data URL feels a bit clunky and not very Pythonic.

Instead of returning a data_url, I think we should return a "file" or something file-like.

Take a look at what [streamlit does](https://docs.streamlit.io/develop/api-reference/widgets/st.file_uploader#:~:text=The%20UploadedFile%20class%20is%20a%20subclass%20of%20BytesIO%2C%20and%20therefore%20it%20is%20%22file%2Dlike%22.%20This%20means%20you%20can%20pass%20them%20anywhere%20where%20a%20file%20is%20expected.). They [subclass Python's io.BytesIO](https://github.com/streamlit/streamlit/blob/0367e69e9fd5ac9fc9fc24a62ed75cf0c0d44a84/lib/streamlit/runtime/uploaded_file_manager.py#L56) which gives it file-like semantics. It's nice because then you can use it in Python APIs like `pd.read_csv(uploaded_file)`

Reply:
Thanks for the pointer. I agree that makes more sense. Initially I didn't realize I could send bytes back to the server, but once you pointed our the proto serialization, then that made it easy. 

I ended up storing the file contents as BytesIO (but didn't subclass the file metadata class as bytesio). So api is slightly more verbose. Would end up being `pd.read_csv(uploaded_file.contents)`.

Reply:
I'm curious why you decided to not subclass? I'm not sure if subclassing BytesIO is hacky or not, but it does make it more convenient to use.

Reply:
I have a preference for composition over inheritance. So in this case it seemed the contents was really the BytesIO part. And the rest was just metadata.

However, I can definitely see the argument for subclassing the uploaded file as well. And I don't have a strong opinion on this either way.

So I've updated it to be a subclass instead.